test/core/security/credentials_test.cc

```diff
@@ -3677,6 +3613,62 @@ TEST(CredentialsTest,
   grpc_channel_credentials_release(composite_creds_2);
 }
 
+TEST(CredentialsTest, TestTlsCredentialsCompareSuccess) {
+  auto* tls_creds_1 =
+      grpc_tls_credentials_create(grpc_tls_credentials_options_create());
+  auto* tls_creds_2 =
+      grpc_tls_credentials_create(grpc_tls_credentials_options_create());
+  EXPECT_EQ(tls_creds_1->cmp(tls_creds_2), 0);
+  EXPECT_EQ(tls_creds_2->cmp(tls_creds_1), 0);
+  grpc_channel_credentials_release(tls_creds_1);
+  grpc_channel_credentials_release(tls_creds_2);
+}
+
+TEST(CredentialsTest, TestTlsCredentialsWithVerifierCompareSuccess) {
+  auto* options_1 = grpc_tls_credentials_options_create();
+  options_1->set_certificate_verifier(
+      MakeRefCounted<HostNameCertificateVerifier>());
+  auto* tls_creds_1 = grpc_tls_credentials_create(options_1);
+  auto* options_2 = grpc_tls_credentials_options_create();
+  options_2->set_certificate_verifier(
+      MakeRefCounted<HostNameCertificateVerifier>());
+  auto* tls_creds_2 = grpc_tls_credentials_create(options_2);
+  EXPECT_EQ(tls_creds_1->cmp(tls_creds_2), 0);
+  EXPECT_EQ(tls_creds_2->cmp(tls_creds_1), 0);
+  grpc_channel_credentials_release(tls_creds_1);
+  grpc_channel_credentials_release(tls_creds_2);
+}
+
+TEST(CredentialsTest, TestTlsCredentialsCompareFailure) {
```

Comment:
To be actually safe here, we need a test that shows that the comparison fails if any one field is different.  Otherwise, we could be subject to serious security bugs.

And this is another thing that would be really nice to automatically update whenever we add a field to `grpc_tls_credentials_options`.

Reply:
Done